### PR TITLE
Feature/add tab view controller

### PR DESCRIPTION
**Pull request by garrincha33**

### CODE REVIEW

File: CryptoLightNews/MainTabBarController.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
+//
+//  MainTabBarController.swift
+//  CryptoLightNews
+//
+//  Created by Richard Price on 15/12/2022.
+//
+
+import UIKit
+
+class MainTabBarController: UITabBarController {
```
Comment (ananogal) [inline CryptoLightNews/MainTabBarController.swift:10]: Your indentation is a little off... You can tackle this by going to Xcode Menu -> Preferences and selecting the TextEditing options. This is a thing. Have a look at this [article](https://medium.com/@peter.lafferty/resources-for-ios-coding-standards-and-styles-7f0badf4581b)

File: CryptoLightNews/MainTabBarController.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        navController.tabBarItem.image = image
+        return navController
+    }
+    
+    private func setupUI() {
+        viewControllers = [
+            setupNavControllers(with: UIViewController(), title: "News", image: #imageLiteral(resourceName: "cryptoNews")),
+            setupNavControllers(with: UIViewController(), title: "Prices", image: #imageLiteral(resourceName: "prices_new")),
+        ]
+        //modify tab bar insets
```
Comment (ananogal) [inline CryptoLightNews/MainTabBarController.swift]: It's not good to write comments in your code. If you need them this means you probably have too much code in your function/method/class. What do you think about moving this code to its own function? The comment really gives you the right clue: `updateTabInsects` can be a candidate

File: CryptoLightNews/SceneDelegate.swift [hunk 3]
```diff
@@ -11,12 +11,12 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
 
     var window: UIWindow?
 
-
     func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
-        // Use this method to optionally configure and attach the UIWindow `window` to the provided UIWindowScene `scene`.
```
Comment (ananogal) [inline CryptoLightNews/SceneDelegate.swift:16]: Please remove the comments

File: CryptoLightNews/ViewController.swift [hunk 4]
```diff
@@ -11,7 +11,7 @@ class ViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
```
Comment (ananogal) [inline CryptoLightNews/ViewController.swift:14]: Remove comments

File: CryptoLightNews/ViewController.swift [hunk 5]
```diff
@@ -11,7 +11,7 @@ class ViewController: UIViewController {
...
+        view.backgroundColor = .orange
```
Comment (ananogal) [inline CryptoLightNews/ViewController.swift]: Are you still using this colour? If not remove it, please.

File: CryptoLightNews/MainTabBarController.swift [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                guard let items = tabBar.items else {return}
+                for item in items {
+                    item.title = ""
+                    item.imageInsets = UIEdgeInsets(top: 2, left: 0, bottom: -2, right: 0)
+                }
+        //setup tab colorScheme
```
Comment (ananogal) [inline CryptoLightNews/MainTabBarController.swift]: So if you need a comment here, maybe you need a method? 😉

File: CryptoLightNews/MainTabBarController.swift [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    private func setupUI() {
+        viewControllers = [
+            setupNavControllers(with: UIViewController(), title: "News", image: #imageLiteral(resourceName: "cryptoNews")),
+            setupNavControllers(with: UIViewController(), title: "Prices", image: #imageLiteral(resourceName: "prices_new")),
+        ]
+        //modify tab bar insets
+                guard let items = tabBar.items else {return}
+                for item in items {
+                    item.title = ""
+                    item.imageInsets = UIEdgeInsets(top: 2, left: 0, bottom: -2, right: 0)
```
Comment (ananogal) [inline CryptoLightNews/MainTabBarController.swift]: When we have constants like this, we tend to have them in a single place that we can use, and it's easier to change. We generally use enumerations to hold these constants. And we declare these enumerations at the beginning of the class. Have a look at this [article](https://www.swiftbysundell.com/articles/powerful-ways-to-use-swift-enums/) to get more context.
In this case, we can have two different approaches(please, feel free to choose one of them):
1 - Create constants of the numbers:
```
  enum Sizes {
    static let tapSpace = 2.0
  }
```
And then you use it like this:
```suggestion
 item.imageInsets = UIEdgeInsets(top: Sizes.tapSpace, left: .zero, bottom: -Sizes.tapSpace, right: .zero)
```
(.zero comes with Swift)

2 - The other way is to set a constant of the inset, like this:
```
  enum Sizes {
    static let tabInsets = UIEdgeInsets(top: 2, left: .zero, bottom: -2, right: .zero)
  }
```
And then use it like this: 
```suggestion
  item.imageInsets = Sizes.tabInsets
```

**Review by ananogal — CHANGES_REQUESTED** [review]
I left some comments for you to look at and make changes.
Please read the articles that I left for you.
Also, I will refuse to review your PRs from now on if you don't put a description of what you have changed. 😉 
Remember that you will be working on a team, and they have to understand the changes that you are making.
If you want to go wild and add a template to this repository, which I encourage you to do, you have [here](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/) a good example. (Maybe you don't need all those ticks 😉 )

File: CryptoLightNews/MainTabBarController.swift (hunk 1 above)
Comment (garrincha33) [reply CryptoLightNews/MainTabBarController.swift:10]: no worries, im in this menu what would you like the identation to be?

Screenshot 2022-12-17 at 12 12 38

Comment (garrincha33) [reply CryptoLightNews/MainTabBarController.swift:10]: have re indented hopefully that should solve the issue

File: CryptoLightNews/MainTabBarController.swift (hunk 6 above)
Comment (garrincha33) [reply CryptoLightNews/MainTabBarController.swift]: moved into function called updateTabColorScheme

File: CryptoLightNews/ViewController.swift (hunk 4 above)
Comment (garrincha33) [reply CryptoLightNews/ViewController.swift:14]: removed

**Comment by garrincha33** [comment]
have made the relevant changes, let me know if anything required.  thanks

File: CryptoLightNews/MainTabBarController.swift [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    }
+    private func updateTabColorScheme() {
+        tabBar.isTranslucent = true
+        tabBar.barTintColor = .clear
+        tabBar.tintColor = .systemGreen
+        UINavigationBar.appearance().prefersLargeTitles = true
+    }
+}
+enum Sizes {
```
Comment (ananogal) [inline CryptoLightNews/MainTabBarController.swift:45]: Sorry, maybe I didn't explain correctly. My fault. 
This enumeration should be inside the class. This way, you can have many of them with the same name, and they will not clash. So when we create the `NewsViewController`, for instance, we can have `Sizes`, too, with different constants, and the name won't clash with this one.
```suggestion
class MainTabBarController: UITabBarController {
   enum Sizes {
   ...
   }
   ...
}
```

**Review by ananogal — APPROVED** [review]
I left a small comment. But in general, it looks really good. Great job! 💪

**Pull request merged**